group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -1099,6 +1101,42 @@ public List<JoinGroupResponseData> joinWithNMembers(
             return joinResponses;
         }
 
+        public CoordinatorResult<LeaveGroupResponseData, Record> sendGenericGroupLeave(
+            LeaveGroupRequestData request
+        ) {
+            RequestContext context = new RequestContext(
+                new RequestHeader(
+                    ApiKeys.LEAVE_GROUP,
+                    ApiKeys.LEAVE_GROUP.latestVersion(),
+                    "client",
+                    0
+                ),
+                "1",
+                InetAddress.getLoopbackAddress(),
+                KafkaPrincipal.ANONYMOUS,
+                ListenerName.forSecurityProtocol(SecurityProtocol.PLAINTEXT),
+                SecurityProtocol.PLAINTEXT,
+                ClientInformation.EMPTY,
+                false
+            );
+
+            return groupMetadataManager.genericGroupLeave(context, request);
+        }
+
+        public void verifyLeaveGroupResponse(
+            LeaveGroupResponseData actualResponse,
+            Errors expectedTopLevelError,
+            List<Errors> expectedMemberLevelErrors
+        ) {
+            assertEquals(expectedTopLevelError.code(), actualResponse.errorCode());
+            if (!expectedMemberLevelErrors.isEmpty()) {
+                assertEquals(expectedMemberLevelErrors.size(), actualResponse.members().size());
+                for (int i = 0; i < expectedMemberLevelErrors.size(); i++) {
+                    assertEquals(expectedMemberLevelErrors.get(i).code(), actualResponse.members().get(i).errorCode());
+                }
+            }
```

Comment:
I am not a fan of this because we don't verify all the fields (e.g. member id) in the response. Should we generate the expected response and use `assertEquals(expected, actual)`?